[TTS] add multi-speaker support for finetuning hifigan vocoder

jerryuhoo opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/PaddleNLP/pull/26 -->
### PR types
New features

### PR changes
Others

### Describe
fix #1268
使hifigan支持多说话人finetune。为了使dump中的train，dev，test和dump_finetune中一致，在gen_gta_mel.py中去掉了根据duration.txt来分配到这三个文件夹中的代码，重新读取了原数据，和preprocess保持一致。
在link_wav.py中加入了进度条，改进了读取utt_id的方式（因为我的一些数据集名称有多个下划线符号），并且是否可以考虑把link_wav.py移动到paddlespeech这个文件夹下面？因为目前支持多说话人finetune的话其他数据集也会用到。
（aishell3和baker的我没做过实际测试，理论上应该可以吧，我用自己的数据集测试了一下可以跑通）

jerryuhoo committed:
add multi-speaker support for finetuning hifigan vocoder

yt605155624 commented:
感觉 link_wav可以放到 ${MAIN_ROOT}/paddlespeech/t2s/exps/gan_vocoder/ 多个 vocoder 可以共用一个

yt605155624 commented on examples/csmsc/voc5/finetune.sh:
```diff
@@ -15,7 +15,10 @@ if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
         --dur-file=durations.txt \
         --output-dir=dump_finetune \
-        --phones-dict=fastspeech2_nosil_baker_ckpt_0.4/phone_id_map.txt
+        --phones-dict=fastspeech2_nosil_baker_ckpt_0.4/phone_id_map.txt \
```
voc3 的 finetune.sh 也要调用 gen_gta_mel.py, 所以也要输入 dataset 和 rootdir，要不应该会报错

jerryuhoo replied:
好的，已增加这两个参数，移除了csmsc的voc5中的speaker-dict。

jerryuhoo committed:
add --dataset and --rootdir to voc3 finetune script

add --dataset and --rootdir to voc3 finetune script, also remove --speaker-dict in voc5.

jerryuhoo committed:
move link_wav.py to paddlespeech/t2s/exps/gan_vocoder/

move link_wav.py to paddlespeech/t2s/exps/gan_vocoder/ so that different vocoders can use one script.

yt605155624 commented on examples/csmsc/voc5/finetune.sh:
```diff
@@ -15,11 +15,13 @@ if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
 
 if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
-    python3 local/link_wav.py \
+    python3 ${MAIN_ROOT}/paddlespeech/t2s/exps/gan_vocoder/link_wav.py \
```
调用 link_wav.py 的时候，参考 synthesize 等，直接用 ${BIN_DIR}/../link_wav.py 即可
${BIN_DIR} 的含义可以看一下  path.sh

zh794390558 commented:
@jerryuhoo @yt605155624 

link_wav这种工具类的文件不建议放到paddelespeech内部.
1.link_wav可以放到utils目录下共用。
2.其他example需要用的话，可以通过soft link（相对路径）的方式链接到对应的local目录下。

yt605155624 commented:
提交 tts 的代码，commit 中可以增加 " , test=tts"
asr " , test=asr"
doc " , test=doc_fix"
跳过不必要的 CI 检查

jerryuhoo committed:
Simplify link_wav.py path

jerryuhoo commented:
> @jerryuhoo @yt605155624
> 
> link_wav这种工具类的文件不建议放到paddelespeech内部. 1.link_wav可以放到utils目录下共用。 2.其他example需要用的话，可以通过soft link（相对路径）的方式链接到对应的local目录下。

好的，我再修改一下

yt605155624 commented:
> @jerryuhoo @yt605155624
> 
> link_wav这种工具类的文件不建议放到paddelespeech内部. 1.link_wav可以放到utils目录下共用。 2.其他example需要用的话，可以通过soft link（相对路径）的方式链接到对应的local目录下。

刚才商量了一下，examples 相关的工具类的东西还是放到 PaddleSpeech 的 utils 里面，因为 paddlespeech 涉及到打包，所以还是不放进去了
@jerryuhoo 放到根目录 PaddleSpeech 的 utils 里面，按照如下的写法，可以通过 python3 link_wav.py 的方式直接调用
https://github.com/PaddlePaddle/PaddleSpeech/blob/f7ffd9917c0434577b8dc941e09df09b59191570/examples/csmsc/voc5/path.sh#L4

zh794390558 commented:
主要是统一风格，形成认知。

jerryuhoo committed:
change link_wav.py path, test=tts

yt605155624 reviewed (approved):
LGTM

yt605155624 committed:
Merge branch 'develop' into develop
